docs/basics.txt

```diff
@@ -635,23 +635,144 @@ bin/gremlin-server.sh ./conf/gremlin-server/http-gremlin-server.yaml
 curl -XPOST -Hcontent-type:application/json -d '{"gremlin":"g.V().count()"}' http://[IP for JanusGraph server host]:8182
 ```
 
+=== JanusGraph Server as Both a WebSocket and HTTP Endpoint
+
+As of JanusGraph 0.2.0, you can configure your gremlin-server.yaml to accept both WebSocket and HTTP connections over the same port. This can be achieved by changing the channelizer in any of the previous examples as follows.
+
+```
+channelizer: org.apache.tinkerpop.gremlin.server.channel.WsAndHttpChannelizer
+```
+
 === Advanced JanusGraph Server Configurations
 
-==== WebSocket versus HTTP
+==== Authentication over HTTP
+There are two ways to authenticate over HTTP with JanusGraph Server: Basic authentication and hash-based messsage authentication code (https://en.wikipedia.org/wiki/Hash-based_message_authentication_code[HMAC]) Auth
 
-JanusGraph Server must be run with a configuration for either WebSocket (for Gremlin Console or other WebSocket programs) or re-configured for HTTP to work with HTTP clients.
+===== HTTP Basic authentication
+To enable Basic authentication in JanusGraph Server include the following configuration in your gremlin-server.yaml.
 
-Currently, there is no way to configure a single running instance of JanusGraph Server to communicate simultaneously with clients talking WebSocket and HTTP.
-However, it is possible to create configuration files (following the steps in this chapter) for both WebSocket and HTTP and start two instances of the JanusGraph Server by pointing it to the various configurations when it is started.
-In this way, the same JanusGraph database instance can be reached through different JanusGraph Servers through both WebSocket and HTTP calls.
-Make sure the port parameter in each yaml configuration is different if starting more than one JanusGraph Server on the same machine:
+```
+ graphs: {
+     credentialGraph: conf/gremlin-server/http-janusgraph-hbase-server.properties
+ }
+ ...
+ authentication: {
+   authenticator: org.janusgraph.graphdb.tinkerpop.gremlin.server.auth.JanusGraphSimpleAuthenticator,
+   authenticationHandler: org.apache.tinkerpop.gremlin.server.handler.HttpBasicAuthenticationHandler,
+   config: {
+     defaultUsername: user,
+     defaultPassword: password,
+     credentialsDb: conf/gremlin-server/http-janusgraph-hbase-server.properties
+   }
+ }
+```
+
+Note that credentialGraph and credentialsDb should point to the same properties file. `credentialGraph` is not required, but make it easier to have the graph available to make changes through the http://tinkerpop.apache.org/docs/3.2.6/reference/#credentials-dsl[TinkerPop CredentialGraph DSL]
+
+Verify that basic authentication is configured correctly. For example
 
 ```
-port: 8182
+curl -v -XPOST http://localhost:8182 -d '{"gremlin": "g.V().count()"}'
+```
+
+should return a 401 if the authentication is configured correctly and
+
+```
+curl -v -XPOST http://localhost:8182 -d '{"gremlin": "g.V().count()"}' -u user:password
```

Comment:
Here I get an error (`No such property: g for class: Script3`). Is there a missing step to get graph queries to work? I can resolve by changing `credentialsGraph` to `graph` but is there another way since you want to keep the credentialsGraph name?

Reply:
@sjudeng I've removed credential graph and added an `IMPORTANT` block in each section stating that the `credentialsDb` should be different from the users `graphs`.

Reply:
Thanks. To avoid confusion can you now also change `conf/gremlin-server/http-janusgraph-hbase-server.properties` here and below to something else to make it clear it should be different? Maybe `conf/janusgraph-credentials-server.properties` or something. Also add documentation indicating that the properties file should define appropriate storage backend (and index backend if necessary) and stating what it will be used for (e.g. store credentials information).

Reply:
Fixed, assuming the wording is okay for you.